### ISSUE TITLE
Feature Request - Mouse free mode

### ISSUE THREAD

**Issue by hammerandtongs** [thebracket/rustrogueliketutorial#98]
It would be nice to have a mouse free targeting mode.

Partially given https://github.com/thebracket/rltk_rs/issues/40 (mouse doesn't work well enough for me to use) but particularly since it would make the game much more playable for me.

The mouse is pretty incompatible for a right handed player using Vim keys but also having to move to the mouse just to do target selection.

Specced as a combination of tabtarget and free aim -

An ability execution activates target mode. Either via number keys, inventory selection or a keypress (t) to activate "free aim" mode (ie for inspection while moving around/out of combat).
Tab starts from the closest viable target.
A targeted mob stays targeted turn by turn.
Tab loops over the viable targets. Shift Tab walks backwards.
The player is the "-1" target ie they aren't the first but are the last+1 or first -1
Direction keys override this selection (useful for aoe abilities) but starting from the most recent tab target OR player if there was no target to start.
Tab returns to target list.
Esc returns the player to move mode.
Esc clears a free target but not an entity.
Death of an entity returns the target to player.
Return executes the ability.

Tab target would be nice to have available even in move mode for a quick view of the mobs.

**Comment by thebracket** [comment]
Sounds good to me. I'll see what I can do in the next week or two. Thanks!

**Comment by oubiwann** [comment]
Yeah, I like the approach used by RLs such as Angband for this; gotten quite used to it, and was hoping to see it here, too :-)
